File: leetcode/java/41.Missing Positive.java
import java.io.*;

class Main{

  static PrintStream ps = System.out;
  
  public static int findMissingPositive(int[] A){
  int n = A.length; 
  boolean[] present = new boolean[n + 1];
  for (int i = 0; i < n; i++) {
  if (A[i] > 0 && A[i] <= n)
    present[A[i]] = true;
  }
  for (int i = 1; i <= n; i++)
    if (!present[i])
      return i;
  return n + 1;
  }

  public static void main(String ...s){
   int[] nums={0,-1,3,1};
   ps.println(findMissingPositive(nums));
  }

}

File: leetcode/kotlin/41.Missing Positive.kt
import java.io.*

internal object Main {
    var ps: PrintStream = System.out
    fun findMissingPositive(A: IntArray): Int {
        val n = A.size
        val present = BooleanArray(n + 1)
        for (i in 0 until n) {
            if (A[i] > 0 && A[i] <= n) present[A[i]] = true
        }
        for (i in 1..n) if (!present[i]) return i
        return n + 1
    }

    fun main(vararg s: String?) {
        val nums = intArrayOf(0, -1, 3, 1)
        ps.println(findMissingPositive(nums))
    }
}